pkg/handlers/app.go

```diff
@@ -53,12 +54,21 @@ func (h *Handler) GetPendingApp(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Carefully now, peek at registry credentials to see if we need to prompt for them
+	hasKurlRegistry, err := registry.HasKurlRegistry()
+	if err != nil {
+		logger.Error(errors.Wrapf(err, "failed to check registry status for pending app %s", papp.Slug))
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Not 100% sure what the response should be here. A 300 is returned when it fails in UploadNewLicense.